LinearRegression/LinearRegressionBackend/MLNeuralNetwork/DataSet.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+using MathNet.Numerics.LinearAlgebra;
+
+using LinearRegressionBackend.DataProvider;
+
+namespace LinearRegressionBackend.MLNeuralNetwork
+{
+    public class DataSet
+    {
+        public Matrix<double> TrainingInput;
+        public Matrix<double> TrainingOutput;
+
+        public double[][] TrainingInputArray
+        {
+            get => Numerical.MatrixToJaggedArray(TrainingInput);
+        }
+        public double[][] TrainingOutputArray
+        {
+            get => Numerical.MatrixToJaggedArray(TrainingOutput);
+        }
+
+        public Matrix<double> TestingInput;
+        public Matrix<double> TestingOutput;
+
+        public double[][] TestingInputArray
+        {
+            get => Numerical.MatrixToJaggedArray(TestingInput);
+        }
+        public double[][] TestingOutputArray
+        {
+            get => Numerical.MatrixToJaggedArray(TestingOutput);
+        }
+
+        public DataSet() { }
+
+        [JsonConstructor]
+        public DataSet(
+            double[][] trainingInputArray,
+            double[][] trainingOutputArray,
+            double[][] testingInputArray,
+            double[][] testingOutputArray)
+        {
+            if (trainingInputArray is not null)
```

Comment:
I'm not sure what to do here, because the `LinearAlgebra` package doesn't support empty matrices or vectors, but these null checks aren't very nice.

Reply:
Is it not enough to check the nullity with the `is not`? We assume that Ali's data converter will be not produce null matrices.